src/go/rpk/pkg/cli/cmd/generate/grafana.go

```diff
@@ -182,12 +189,14 @@ func (rowSet *RowSet) processRows(metricFamilies map[string]*dto.MetricFamily) {
 	for _, name := range names {
 		family := metricFamilies[name]
 		var panel graf.Panel
-		if family.GetType() == dto.MetricType_COUNTER {
-			panel = newCounterPanel(family)
+		// hack around redpanda_storage_* metrics: these should be gauge
+		// panels but the metrics type come as COUNTER
+		if family.GetType() == dto.MetricType_COUNTER && !strings.Contains(name, "redpanda_storage") {
```

Comment:
Should this be !strings.HasPrefix?

Also, is the counter vs. gauge thing a redpanda issue?

Reply:
Yeah this looks like a bug. I think all of the repdanda_storage metrics in public_metrics should be gauges. Filed: https://github.com/redpanda-data/redpanda/issues/6316